lib/modules/dosomething/dosomething_global/dosomething_global.module

```diff
@@ -651,6 +651,24 @@ function dosomething_global_url($path, $options = NULL) {
 }
 
 /**
+ * Build a global node URL that adds the prefix and a fragment
+ *
+ * @param string $nid
+ *    The nid of the node
+ * @param string $language
+ *    The language of the URL
+ * @param string $fragment
+ *    Optional fragment to append
+ */
+function dosomething_global_node_url($nid, $language, $fragment = NULL) {
+  $raw_url = 'node/' . $nid;
+  if (isset($fragment)) {
+    $raw_url .= '#' . $fragment;
```

Comment:
Well, we're using `l()` function a lot. And `l()` uses `url()` under the hood. That's why I think for we need to make `url()` reliable as oppose to generate links by our own.
What do you think @angaither?